tests/application/usecases/manufacturer/manufacturer.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { describe, expect, it, vi } from "vitest";
+import { AppError } from "../../../../src/application/errors/errors";
+import { AddManufacturerUseCase } from "../../../../src/application/usecases/manufacturer";
+import { Manufacturer } from "../../../../src/domain/entities/manufacturer";
+import { Logging } from "../../../../src/domain/logger";
+import { AddManufacturer } from "../../../../src/domain/usecases/add-manufacturer-use-case";
+
+const logger: Logging = {
+  error: vi.fn(),
+  info: vi.fn(),
+  warn: vi.fn(),
+  debug: vi.fn(),
+  log: vi.fn(),
+};
+
+const manufacturerRepositoryMock = {
+  create: vi.fn(),
+};
+
+const makeSut = (): { sut: AddManufacturer } => {
+  const sut = new AddManufacturerUseCase(manufacturerRepositoryMock, logger);
+  return { sut };
+};
+
+describe("AddManufacturerUseCase", () => {
+  it("should create a manufacturer successfully", async () => {
+    const { sut } = makeSut();
+    const input: Manufacturer = {
+      name: "Fabricante XYZ",
+      city: "São Paulo",
+      email: "contato@fabricantexyz.com",
+      number: "123",
+      phone: "+55 (11) 98765-4321",
+      state: "SP",
+      street: "Avenida Paulista",
+      website: "https://www.fabricantexyz.com",
+      zip_code: "01310-000",
+    };
+    const resultManufacturer: Manufacturer = {
+      ...input,
+    };
+    manufacturerRepositoryMock.create.mockResolvedValueOnce(resultManufacturer);
+    const result = await sut.execute(input);
+    expect(result).toEqual(resultManufacturer);
+    expect(manufacturerRepositoryMock.create).toHaveBeenCalledWith(input);
```

Comment:
vamos simular a sequencia que ocorre o teste, primeiro que ele seja chamado com o valor esperado
e por fim que retorne o esperado, só alterando a sequência:

```suggestion
    expect(manufacturerRepositoryMock.create).toHaveBeenCalledWith(input);
    expect(result).toEqual(resultManufacturer);
```

Reply:
perfeito